Specifying `[[bin]]` with required features should be a required instruction

The quickstart says one can compile the C library and generate the headers with the following commands: https://github.com/getditto/safer_ffi/blob/b4c13c72c61fffd9b6cc4b8b17b60da556d5eebd/guide/src/introduction/quickstart.md?plain=1#L106-L113

However the quickstart `Cargo.toml` does not include a `[[bin]]` section: https://github.com/getditto/safer_ffi/blob/b4c13c72c61fffd9b6cc4b8b17b60da556d5eebd/guide/src/introduction/quickstart.md?plain=1#L20-L46

This is a problem because `cargo` will automatically compile all binaries and libraries if no selection is given:

```shell
% cargo --version 
cargo 1.71.0 (cfd3bbd8f 2023-06-08)
% cargo help build | grep -i 'target selection' -A2 -m1
       When no target selection options are given, cargo build will build all
       binary and library targets of the selected packages. Binaries are
       skipped if they have required-features that are missing.
% cargo build
   Compiling syn v1.0.109
   Compiling libc v0.2.147
   Compiling unwind_safe v0.1.0
   Compiling scopeguard v1.2.0
   Compiling macro_rules_attribute v0.1.3
   Compiling prettyplease v0.1.25
   Compiling ext-trait-proc_macros v1.0.1
   Compiling with_builtin_macros-proc_macros v0.0.3
   Compiling safer_ffi-proc_macros v0.1.2
   Compiling ext-trait v1.0.1
   Compiling with_builtin_macros v0.0.3
   Compiling extension-traits v1.0.1
   Compiling uninit v0.5.1
   Compiling safer-ffi v0.1.2
   Compiling kriswuollett-examples-safer-c-callback v0.1.0 (/Users/kris/code/kriswuollett/examples-rust/safer-c-callback)
error[E0425]: cannot find function `generate_headers` in crate `kriswuollett_examples_safer_c_callback`
 --> safer-c-callback/src/bin/generate-headers.rs:2:47
  |
2 |     ::kriswuollett_examples_safer_c_callback::generate_headers()
  |                                               ^^^^^^^^^^^^^^^^ not found in `kriswuollett_examples_safer_c_callback`

For more information about this error, try `rustc --explain E0425`.
error: could not compile `kriswuollett-examples-safer-c-callback` (bin "generate-headers") due to previous error
```

Therefore the `[[bin]]` section should be considered as required, rather than optional for improving error messages, and moved to the quickstart:

https://github.com/getditto/safer_ffi/blob/b4c13c72c61fffd9b6cc4b8b17b60da556d5eebd/guide/src/usage/lib-rs.md?plain=1#L84-L94

Which would solve the issue:

```shell
% cargo build
   Compiling proc-macro2 v1.0.66
   Compiling unicode-ident v1.0.11
   Compiling syn v1.0.109
   Compiling paste v1.0.14
   Compiling prettyplease v0.1.25
   Compiling macro_rules_attribute-proc_macro v0.1.3
   Compiling libc v0.2.147
   Compiling safer-ffi v0.1.2
   Compiling scopeguard v1.2.0
   Compiling unwind_safe v0.1.0
   Compiling macro_rules_attribute v0.1.3
   Compiling quote v1.0.31
   Compiling ext-trait-proc_macros v1.0.1
   Compiling with_builtin_macros-proc_macros v0.0.3
   Compiling safer_ffi-proc_macros v0.1.2
   Compiling ext-trait v1.0.1
   Compiling with_builtin_macros v0.0.3
   Compiling extension-traits v1.0.1
   Compiling uninit v0.5.1
   Compiling kriswuollett-examples-safer-c-callback v0.1.0 (/Users/kris/code/kriswuollett/examples-rust/safer-c-callback)
    Finished dev [unoptimized + debuginfo] target(s) in 7.31s
```

--- a/README.md
+++ b/README.md
@@ -63,6 +63,10 @@ crate-type = [
     "lib",        # For downstream Rust dependents: `examples/`, `tests/` etc.
 ]
 
+[[bin]]
+name = "generate-headers"
+required-features = ["headers"]  # Do not build unless generating headers.
+
 [dependencies]
 # Use `cargo add` or `cargo search` to find the latest values of x.y.z.
 # For instance:

--- a/guide/src/introduction/quickstart.md
+++ b/guide/src/introduction/quickstart.md
@@ -32,6 +32,10 @@ crate-type = [
     "lib",        # For downstream Rust dependents: `examples/`, `tests/` etc.
 ]
 
+[[bin]]
+name = "generate-headers"
+required-features = ["headers"]  # Do not build unless generating headers.
+
 [dependencies]
 # Use `cargo add` or `cargo search` to find the latest values of x.y.z.
 # For instance:

--- a/guide/src/usage/lib-rs.md
+++ b/guide/src/usage/lib-rs.md
@@ -81,18 +81,6 @@ file(name), or into the given `Write`-able / "write sink":
 
     to generate the headers.
 
-  - You may also want to add:
-
-    ```toml
-    # Cargo.toml
-    [[bin]]
-    name = "generate-headers"
-    required-features = ["headers"]
-    ```
-
-    to your `Cargo.toml` to improve the error messages when the feature is
-    missing.
-
 <details>
 <summary>More advanced example (runtime-dependent header output)</summary>
 

